src/main/java/dev/blaauwendraad/masker/json/ByteTrie.java

```diff
@@ -28,21 +33,23 @@
  * <p> And lastly, we can make a small optimization to remember all the distinct lengths of the target keys, so that
  * we can fail fast if the incoming key is not of the same length.
  */
+@ParametersAreNonnullByDefault
 final class ByteTrie {
-    private static final int MAX_BYTE_SIZE = 128;
+    private static final int BYTE_OFFSET = -1 * Byte.MIN_VALUE;
+    private final JsonMaskingConfig config;
     private final TrieNode root;
-    private final boolean caseInsensitive;
     private final boolean[] knownByteLengths = new boolean[256]; // byte can be anywhere between 0 and 256 length
 
-    public ByteTrie(boolean caseInsensitive) {
-        this.caseInsensitive = caseInsensitive;
+    public ByteTrie(JsonMaskingConfig config) {
+        this.config = config;
         this.root = new TrieNode();
     }
 
     /**
      * Inserts a word into the trie.
      */
-    public void insert(String word) {
+    public void insert(String word, boolean negativeMatch) {
```

Comment:
Can you add parameters to the JavaDoc? Especially to figure out what `negativeMatch` is I had to read the method implementation

Reply:
Done